pkg/gatherers/workloads/gather_workloads_runtime_infos.go

```diff
@@ -0,0 +1,328 @@
+package workloads
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"hash"
+	"sync"
+	"time"
+
+	appsv1client "k8s.io/client-go/kubernetes/typed/apps/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachinerywait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/klog/v2"
+)
+
+var (
+	labelSelector = "app.kubernetes.io/name=insights-operator-runtime"
+)
+
+// Internal representation of workload infor returned by the insights-operator-runtime component.
+type insightsWorkloadRuntimeInfo struct {
+	OSReleaseID            string             `json:"os-release-id,omitempty"`
+	OSReleaseVersionID     string             `json:"os-release-version-id,omitempty"`
+	RuntimeKind            string             `json:"runtime-kind,omitempty"`
+	RuntimeKindVersion     string             `json:"runtime-kind-version,omitempty"`
+	RuntimeKindImplementer string             `json:"runtime-kind-implementer,omitempty"`
+	Runtimes               []RuntimeComponent `json:"runtimes,omitempty"`
+}
+
+func gatherWorkloadRuntimeInfos(
+	ctx context.Context,
+	h hash.Hash,
+	coreClient corev1client.CoreV1Interface,
+	appClient appsv1client.AppsV1Interface,
+	restConfig *rest.Config,
+) (workloadRuntimes, error) {
+	start := time.Now()
+
+	workloadRuntimeInfos := make(workloadRuntimes)
+
+	klog.Infof("Deploying insights-operator-runtime...\n")
```

Comment:
as mentioned in the comment above, this could be changed so that the `insights-operator-runtime` daemonset is deployed when the operator is installed instead of doing it dynamically